Source/Core/Core/IOS/Network/KD/NetKDRequest.cpp

```diff
@@ -193,51 +193,39 @@ NWC24::ErrorCode NetKDRequestDevice::KDDownload(const u16 entry_index,
   if (!response)
   {
     ERROR_LOG_FMT(IOS_WC24, "Failed to request data at {}", url);
-    return NWC24::WC24_ERR_NETWORK;
+    return NWC24::WC24_ERR_SERVER;
   }
 
-  // Write buffer to temporary file
-  const auto temp = m_ios.GetFS()->CreateAndOpenFile(
-      PID_KD, PID_KD, DL_CNT_PATH, {FS::Mode::ReadWrite, FS::Mode::ReadWrite, FS::Mode::ReadWrite});
-  if (!temp)
+  // Check if the filesize is smaller than the header size.
+  if (response->size() <= sizeof(NWC24::WC24File))
```

Comment:
Did you mean `response->size() < sizeof(NWC24::WC24File)` here? Is a file that's exactly `== sizeof(NWC24::WC24File)` automatically invalid?